var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -0,0 +1,289 @@
+    coupling between packages in-core, serialization, and I/O tasks."""
+
+    homepage = "http://software.llnl.gov/conduit"
+    url = "https://github.com/LLNL/conduit/archive/v0.2.0.tar.gz"
+
+    version('0.2.0', 'd595573dedf55514c11d7391092fd760')
+
+    version('github-master', git='https://github.com/LLNL/conduit.git')
+
+    ###########################################################################
+    # package variants
+    ###########################################################################
+
+    variant("shared", default=True, description="Build Conduit as shared libs")
+
+    variant("cmake", default=True,
+            description="Build CMake (if off, attempt to use cmake from PATH)")
+
+    # variants for python support
+    variant("python", default=True, description="Build Conduit Python support")
+
+    # variants for i/o
+    variant("mpi", default=True, description="Build Conduit MPI Support")
+    variant("hdf5", default=True, description="Build Conduit HDF5 support")
+    variant("silo", default=True, description="Build Conduit Silo support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=True,   description="Build Conduit's docs")
+
+    ###########################################################################
+    # package dependencies
+    ###########################################################################
+
+    #######################
+    # CMake
+    #######################
+    depends_on("cmake@3.3.1", when="+cmake")
+
+    #######################
+    # Python
+    #######################
+    depends_on("python", when="+python")
+    depends_on("py-numpy~blas~lapack", when="+python")
+    depends_on("py-sphinx", when="+python+docs")
+    depends_on("doxygen", when="+docs")
+
+    #######################
+    # I/O Packages
+    #######################
+    depends_on("hdf5+shared~cxx~mpi~fortran", when="+shared")
+    depends_on("hdf5~shared~cxx~mpi~fortran", when="~shared")
+
+    depends_on("silo+shared~fortran", when="+shared")
+    depends_on("silo~shared~fortran", when="~shared")
+
+    #######################
+    # MPI
+    #######################
+    depends_on("mpi", when="+mpi")
+
+    def install(self, spec, prefix):
+        """
+        Build and install Conduit.
+        """
+        with working_dir('spack-build', create=True):
+            host_cfg_fname = self.create_host_config(spec, prefix)
+            cmake_args = []
+            # if we have a static build, we need to avoid
+            # any of spack's default cmake settings related to
+            # rpath
+            if "+shared" in spec:
+                cmake_args.extend(std_cmake_args)
+            else:
+                for arg in std_cmake_args:
+                    if arg.count("RPATH") == 0:
+                        cmake_args.append(arg)
+            cmake_args.extend(["-C", host_cfg_fname, "../src"])
+            cmake(*cmake_args)
+            make()
+            make("install")
```

Comment:
> how much would it save me?

It's not about saving the package author; but about making things standard in a way that's more maintainable down the line.

It looks like this is a good solution to #2658?  If that's the case, then we should put it in `CMakePackage`, activated on the condition it's running on bgq.  It should not go in this PR: because it could mess up people NOT running on bgq, and because it won't fix the problem for bgq users of all the other CMake-based packages in our repo.

BTW... are shared libraries possible?